lib/timers.js

```diff
@@ -60,13 +61,14 @@ const {
 
 // Remove a timer. Cancels the timeout and resets the relevant timer properties.
 function unenroll(item) {
+  if (item._destroyed)
```

Comment:
I'm not sure this is 'correct'...

At very least, `unenroll()` should _always_ call `L.remove(item)`.

That being said, the definitive check for if something is an active timer has been (roughly) kept to `item._idleTimeout > 0 && typeof item._idleCallback === 'function'` for some time (partially since inception), and I think we should avoid changing that assumption if possible?

I feel like the old behavior was more foolproof.

Reply:
I mean, the goal is to establish a lifecycle where `_destroyed` or something else can sufficiently indicate a timer/object that doesn't need to have anything done to it. It's possible we can (and should) just do it with the check you mentioned.

In addition, the notion of there being an undefined `kRefed` and `null` one is very complex and also means the state of the timer is lossy. I think it's important to get away from that current implementation and towards something where kRefed doesn't need to be unset when a timer expires.